core/config/solver_config.hpp

```diff
@@ -17,19 +19,18 @@ namespace config {
 
 
 template <typename SolverParam>
-inline void common_solver_parse(SolverParam& params, const pnode& config,
-                                const registry& context,
-                                type_descriptor td_for_child)
+void common_solver_parse(SolverParam& params, config_decorator& decorator,
```

Comment:
I'm wondering if we should just call this parameter `config` itself, since it acts like the original config parameter, plus a little bit extra